pkg/v1/cli/carvelhelpers/package.go

```diff
@@ -37,5 +40,12 @@ func ProcessCarvelPackage(image string) ([]byte, error) {
 		return nil, errors.Wrap(err, "error while saving file")
 	}
 
-	return ResolveImagesInPackage([]string{f.Name()})
+	inputFilesForImageResolution := []string{f.Name()}
+
+	imgpkgDir := filepath.Join(pkgDir, ".imgpkg")
```

Comment:
not entirely the scenarios we need to handle. I this suggesting that the imgpkgDir variable can be pointing to either a directory or file. If so, some unit test cases verifying both scenarios would have clarified the implementation quite a bit. (Also the variable name is rather misleading if so).

Reply:
Added comment above the lines with explanation. 
```
	// Each package contains `config` and `.imgpkg` directory
	// `config` directory contains ytt files
	// `.imgpkg` directory contains ImageLock configuration for ImageResolution
```

Reply:
Renames the function below as `PathExists` as it applies to file as well as directories.